### PR TITLE
Chore: previous year tests

### PR DESCRIPTION
**Pull request by dleard**
This needed a new trigger to restrict old v1 applications in draft from being submitted in subsequent years.
 - [x] new trigger to restrict old submissions
 - [x] refactor revision mutation to allow for older submissions (fix use of active form_jsons) 
 - [x] pgTap tests ensuring older applications (as long as a v1 was submitted) can be re-worked in subsequent years
 - [x] cypress tests ensuring our UI allows for re-working older applications (when allowed)

### CODE REVIEW

File: schema/deploy/trigger_functions/restrict_old_application_submission.sql [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+-- requires: tables/application_revision_status
+
+begin;
+
+create or replace function ggircs_portal_private.restrict_old_application_submission()
+  returns trigger as $$
+    begin
+      if (select reporting_year from ggircs_portal.opened_reporting_year()) is not null
+        and new.version_number = 1
+        and (select reporting_year from ggircs_portal.application where id = new.application_id) < (select reporting_year from ggircs_portal.opened_reporting_year()) then
```
Comment (matthieu-foucault) [inline schema/deploy/trigger_functions/restrict_old_application_submission.sql:12]:
```suggestion
        and (select reporting_year from ggircs_portal.application where id = new.application_id) != (select reporting_year from ggircs_portal.opened_reporting_year()) then
```

File: schema/deploy/trigger_functions/restrict_old_application_submission.sql [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+          if (new.application_revision_status = 'submitted')
+            and (select application_revision_status from ggircs_portal.application_revision_status where application_id = new.application_id order by id desc limit 1) = 'draft'
+          then
+            raise exception 'You cannot submit a first version of an application for a previous year';
+          end if;
+          if (new.application_revision_status = 'draft') then
+            raise exception 'You cannot start a draft application for a previous year';
+          end if;
```
Comment (matthieu-foucault) [inline schema/deploy/trigger_functions/restrict_old_application_submission.sql:20]: Isn't there another trigger checking this? Or was it not covered by the other trigger?

File: schema/test/unit/mutations/create_application_revision_mutation_chain_test.sql [hunk 3]
```diff
@@ -17,7 +17,7 @@ $fun$
   select mocks.set_mocked_time_in_transaction('2021-04-01 14:49:54.191757-07'::timestamptz - interval '1 second');
 $fun$language sql;
 
-select plan(9);
+select * from no_plan();
```
Comment (matthieu-foucault) [inline schema/test/unit/mutations/create_application_revision_mutation_chain_test.sql:20]: create a test plan?

**Review by matthieu-foucault — CHANGES_REQUESTED** [review]
Do we need any client-side changes, to hide the submit button for drafts of a previous year?

File: schema/deploy/trigger_functions/restrict_old_application_submission.sql (hunk 2 above)
Comment (dleard) [reply schema/deploy/trigger_functions/restrict_old_application_submission.sql:20]: `ensure_window_open_submit_application_status` is similar, it blocks drafts from being created outside the opened reporting window.
This check ensures that new draft applications for older years can't be started during another year's open window.

**Comment by dleard** [comment]
> Do we need any client-side changes, to hide the submit button for drafts of a previous year?

I know we hide the 'start application' button on the facilities page for drafts that were not submitted from a previous year, but I bet the submit button is still there if you went to the submission page directly. I'll take a look and add some logic to hide it if we haven't handled that.

**Review by matthieu-foucault — APPROVED** [review]

**Pull request merged**